src/Teams/Data/IRepository.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Teams.Data
+{
+    interface IRepository<T> where T : class
+    {
+        IQueryable<T> GetAll();
+        T GetById { get; set;}
```

Comment:
How are you planning to retrieve entity with that property? Can you please add a code example how it will look like?

Reply:
Also here I believe better to return Task<T>, inside to find will be used async method